### PR TITLE
Add Takeoff step Context

### PR DESCRIPTION
**Pull request by timvancann**
## Summary:
This change allows steps to share specific context values with other steps.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] There is no commented out code in this PR.
  - [x] There are no new TODOs in this PR

If exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated

### CODE REVIEW

File: takeoff/context.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+from enum import Enum, auto, unique
+from typing import Any, Dict
+
+
+@unique
+class ContextKey(Enum):
+    EVENTHUB_PRODUCER_SECRETS = auto()
```
Comment (danielvdende) [inline takeoff/context.py]: probably good to emphasise this is for producer *policies*

File: takeoff/azure/deploy_to_kubernetes.py [hunk 2]
```diff
@@ -151,6 +152,7 @@ def _get_docker_registry_secret(self) -> str:
     def _render_kubernetes_config(
         self, kubernetes_config_path: str, application_name: str, secrets: Dict[str, str]
     ) -> str:
+        print(secrets)
```
Comment (danielvdende) [inline takeoff/azure/deploy_to_kubernetes.py]: stray print ;-)

File: takeoff/context.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class Context(metaclass=Singleton):
+    def __init__(self):
+        self.context = {}
```
Comment (danielvdende) [inline takeoff/context.py]: I don't think I'd call this variable context. Seems odd considering the classname is the same (with different capitalisation). perhaps `data` ?

File: takeoff/context.py (hunk 1 above)
Comment (timvancann) [reply takeoff/context.py]: 👍

File: takeoff/context.py (hunk 3 above)
Comment (timvancann) [reply takeoff/context.py]: Done, also made it private.

File: docs/_deployment-steps/deploy_to_kubernetes.md [hunk 4]
```diff
@@ -129,3 +129,29 @@ steps:
     
   restart_unchanged_resources: true
 ```
+
+### Eventhub producer policy secrets
+Eventhub producer policy secrets from [`configure_eventhub`](deployment-step/configure-eventhub) are available during this task. This make it possible for the configuration below:
```
Comment (danielvdende) [inline docs/_deployment-steps/deploy_to_kubernetes.md]: the last sentence is a bit weird here ;-)

**Pull request merged**